basyx.submodelrepository/basyx.submodelrepository-backend-inmemory/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/InMemorySubmodelRepository.java

```diff
@@ -235,6 +243,82 @@ public OperationVariable[] invokeOperation(String submodelId, String idShortPath
 	}
 
 
+	@Override
+	public java.io.File getFileByPathSubmodel(String submodelId, String idShortPath) {
+		throwIfSubmodelDoesNotExist(submodelId);
+		SubmodelElement submodelElement = submodelServices.get(submodelId).getSubmodelElement(idShortPath);
+
+		throwIfSmElementIsNotAFile(submodelElement);
+
+		File fileSmElement = (File) submodelElement;
+		String filePath = fileSmElement.getValue();
+
+		throwIfFileDoesNotExist(fileSmElement, filePath);
+
+		return new java.io.File(filePath);
+
+	}
+
+	@Override
+	public void setFileValue(String submodelId, String idShortPath, InputStream inputStream) throws IOException {
+		throwIfSubmodelDoesNotExist(submodelId);
+
+		SubmodelElement submodelElement = submodelServices.get(submodelId).getSubmodelElement(idShortPath);
+		throwIfSmElementIsNotAFile(submodelElement, inputStream);
```

Comment:
Yes, this could happen because the REST request accepts **idShortPath** only as a parameter. Hence, this path could be for any kind of SME. Of course, this **idShortPath** should be of File SME, but validation is necessary, so this check is important.
Does this answer your question?